src/browser_action/vitals.js

```diff
@@ -224,53 +233,144 @@
+                  'Time (ms)': metric.value.toFixed(2),
+                },
+                {
+                  'LCP Breakdown': 'timeToFirstByte',
+                  'Time (ms)': metric.attribution.timeToFirstByte.toFixed(2),
+                },
+                {
+                  'LCP Breakdown': 'resourceLoadDelay',
+                  'Time (ms)': metric.attribution.resourceLoadDelay.toFixed(2),
+                },
+                {
+                  'LCP Breakdown': 'resourceLoadTime',
+                  'Time (ms)': metric.attribution.resourceLoadTime.toFixed(2),
+                },
+                {
+                  'LCP Breakdown': 'elementRenderDelay',
+                  'Time (ms)': metric.attribution.elementRenderDelay.toFixed(2),
+                }
+              ]
+            )
+          }
+        }
         break;
       case "CLS":
-        // CLS has a startTime, but not a duration.
-        // Could visualize the time between rendering tasks (Commit-to-Commit).
-        // Skip for now.
+        if (enableConsoleTables) {
+          // Add a nice console output of all the shifts
+          let entries = [{'CLS Breakdown': 'CLS', 'Time (ms)': metric.value}]
+          metric.entries.map((entry) => {
+            entries.push({'CLS Breakdown': entry.entryType, 'Time (ms)': entry.value});
+          });
+          console.table(entries)
+        }
         break;
       case "INP":
-        if (metric.entries.length > 0) {
-          const inpEntry = metric.entries[0];
+        if (metric.attribution && metric.attribution.eventEntry) {
+          const inpEntry = metric.attribution.eventEntry;
 
           // RenderTime is an estimate, because duration is rounded, and may get rounded keydown
           // In rare cases it can be less than processingEnd and that breaks performance.measure().
           // Lets make sure its at least 4ms in those cases so you can just barely see it.
           const presentationTime = inpEntry.startTime + inpEntry.duration;
           const adjustedPresentationTime = Math.max(inpEntry.processingEnd + 4, presentationTime);
 
-          performance.measure(`[Web Vitals] INP.duration (${inpEntry.name})`, {
-            start: inpEntry.startTime,
-            end: presentationTime,
-          });
-          performance.measure(`[Web Vitals] INP.inputDelay (${inpEntry.name})`, {
-            start: inpEntry.startTime,
-            end: inpEntry.processingStart,
-          });
-          performance.measure(`[Web Vitals] INP.processingTime (${inpEntry.name})`, {
-            start: inpEntry.processingStart,
-            end: inpEntry.processingEnd,
-          });
-          performance.measure(`[Web Vitals] INP.presentationDelay (${inpEntry.name})`, {
-            start: inpEntry.processingEnd,
-            end: adjustedPresentationTime,
-          });
+          if (enableUserTiming) {
+            performance.measure(`[Web Vitals Extension] INP.duration (${inpEntry.name})`, {
```

Comment:
Aside: any reason these INP sub-parts aren't a standard part of `web-vitals` attribution?

Reply:
It's a good question!

Maybe @mmocny can confirm but maybe because it was less mature back then? Or because they are directly referenceable from the Entry itself it wasn't as important to add to the web-vitals build, as that differs slightly from LCP which needs the nav entry for TTFB and then the LCP resource time entry for the other parts so a bit more complicated.

Anyway, as you know Michael's been iterating on this recently so I imagine these sub-parts could change when he adds his "live breakdown" piece after this PR is merged. But still think this is a good first step and good to include for the reasons given above.

But do agree we should add whatever we finally land on to the web-vitals attribution build so other users can grab them easily - at which point we can reference them here too. It wouldn't really simplify the `performance.measure` (which needs both entries anyway) but could simplify the table code slightly (which only needs the duration).

Reply:
We did want to add something to web-vitals attribution, but I decided to wait until we knew better exactly what we want to recommend measuring so as not to break compat eventually.

A long time ago we recommended just doing what we do here in the extension (split a single event into the three components: input-delay, processing-duration, presentation-delay)-- but as Barry says you can do that easily from a single entry.

In practice, processing-duration is better to SUM(all-events) in a single Animation Frame, rather than just looking at a single event.  But when you do that, you end up having to re-define input-delay and presentation-delay to be "delay until first event" and "delay after last event".

Finally, presentation-delay can be split into "rendering-duration" (on main) and "presentation-delay" (off main).

All these things become easier (or possible, even) with Long Animation Frame (LoAF) reporting, and so that's where I've been iterating.


(The direction tooling is going is to "squash" multiple interactions in a single frame down to a single frame latency-- so we'll probably want to evolve this in the extension as well.  I'll try a stab at updating INP user timings after updating INP logs.)